### PR TITLE
BulletType despawnUnit field

### PR DESCRIPTION
**Pull request by SMOLKEYS**
Nothing to say.

---

If your pull request is **not** translation or serverlist-related, read the list of requirements below and check each box:

- [X] I have read the [contribution guidelines](https://github.com/Anuken/Mindustry/blob/master/CONTRIBUTING.md).
- [X] I have ensured that my code compiles, if applicable.
- [ ] I have ensured that any new features in this PR function correctly in-game, if applicable.

### CODE REVIEW

**Comment by Mythril382** [comment]
intervalUnit when

File: core/src/mindustry/entities/bullet/BulletType.java [hunk 1]
```diff
@@ -443,6 +445,10 @@ public void despawned(Bullet b){
             createFrags(b, b.x, b.y);
         }
 
+        if(despawnUnit != null){
+            despawnUnit.spawn(b.team, b.x + Mathf.range(2f), b.y);
```
[the remark this answers is not in the document]
Comment (SMOLKEYS) [reply core/src/mindustry/entities/bullet/BulletType.java:449]: It's to essentially prevent "unit stacking", where ground units could stack on top of each other in one point.

[the remark this answers is not in the document]
Comment (SMOLKEYS) [reply core/src/mindustry/entities/bullet/BulletType.java:449]: (There are 15 maces stacked on top of each other in this image)

**Comment by Anuken** [comment]
Seems excessive, I think a random spread of even 0.1 should work.

**Pull request merged**